### PR TITLE
[DateTimePicker] Fix toolbar time order when `theme.rtl=true`

### LINKED ISSUE #5561: [pickers] Time in DateTimePickerToolbar is backwards in RTL theme
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

It should display the time correctly (hh:mm)

### Expected behavior 🤔

It displays the time in reverse (mm:hh)

### Steps to reproduce 🕹

Link to live example (2 live examples):
- go to docs [DateTimePicker mobile example](https://mui.com/x/react-date-pickers/date-time-picker/#responsiveness) and change to RTL in options.
- [sandbox](https://codesandbox.io/s/responsivedatetimepickers-demo-mui-x-forked-pwdtlk?file=/demo.tsx). copied sandbox from DOCS and added RTL ([guide in docs](https://mui.com/material-ui/guides/right-to-left/))

Steps (same for the above live examples):
1. Create MobileDateTimePicker
2. add RTL
3. Open it

### Context 🔦

Im trying to use MobileDateTimePicker with RTL. the time seems to be revesered.

I saw the same issue in the old Material UI [here](https://github.com/mui/material-ui-pickers/issues/1071). Also, I saw code that is supposed to handle this in the new Mui [TimePickerToolbar.tsx#L30](https://github.com/mui/mui-x/blob/0a0235b7b47d7854f79e54d3c8ce8ec3c42b3d18/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx#L30). but somehow the bug still exists

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  System:
    OS: Linux 5.13 Ubuntu 20.04.4 LTS (Focal Fossa)
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: Not Found
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
  Browsers:
    Chrome: 101.0.4951.41
  npmPackages:
    @emotion/react: ^11.7.1 => 11.9.3 
    @emotion/styled: ^11.6.0 => 11.9.3 
    @mui/base:  5.0.0-alpha.86 
    @mui/icons-material: ^5.4.1 => 5.8.4 
    @mui/lab: ^5.0.0-alpha.70 => 5.0.0-alpha.87 
    @mui/material: ^5.2.6 => 5.8.5 
    @mui/private-theming:  5.8.4 
    @mui/styled-engine:  5.8.0 
    @mui/styles: ^5.2.3 => 5.8.4 
    @mui/system:  5.8.5 
    @mui/types:  7.1.4 
    @mui/utils:  5.8.4 
    @mui/x-date-pickers: ^5.0.0-beta.1 => 5.0.0-alpha.1 
    @types/react: ^17.0.11 => 17.0.47 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^4.3.4 => 4.7.4
```
</details>

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix #5561

### CODE REVIEW

File: packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx [hunk 1]
```diff
@@ -26,12 +26,12 @@ export interface DateTimePickerToolbarProps<TDate>
...
 
-const useUtilityClasses = (ownerState: DateTimePickerToolbarProps<any>) => {
-  const { classes } = ownerState;
+const useUtilityClasses = (ownerState: DateTimePickerToolbarProps<any> & { theme: Theme }) => {
+  const { classes, theme } = ownerState;
   const slots = {
     root: ['root'],
     dateContainer: ['dateContainer'],
-    timeContainer: ['timeContainer'],
+    timeContainer: ['timeContainer', theme.direction === 'rtl' && 'timeLabelReverse'],
```
[the remark this answers is not in the document]
Comment (alexfauquette) [reply packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx:34]: Agree, but it's to be consistent in the way we handle that with TimePicker toolbar
https://github.com/mui/mui-x/blob/0a0235b7b47d7854f79e54d3c8ce8ec3c42b3d18/packages/x-date-pickers/src/TimePicker/TimePickerToolbar.tsx#L30

[the remark this answers is not in the document]
Comment (LukasTy) [reply packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx:34]: Fair point. 🤔

**Review by LukasTy — APPROVED** [review]

**Review by flaviendelangle — APPROVED** [review]

**Comment by Eliav2** [comment]
`Fix toolbar time order when theme.rtl=true`, could you elaborate? what `theme.rtl`? mui theme provider does not support `rtl` propery

**Comment by LukasTy** [comment]
> `Fix toolbar time order when theme.rtl=true`, could you elaborate? what `theme.rtl`? mui theme provider does not support `rtl` propery

Thank you for bringing this up! 🙏 
I've updated the PR title to be more precise.
We meant to say `theme.direction.rtl` instead of `theme.rtl`. 😉

**Comment by alexfauquette** [comment]
@LukasTy Do we have a docs section about RTL?

I found [this](https://mui.com/x/react-date-pickers/base-concepts/#field-components) about extra unicode characters, but no section about how to setup RTL like for the data grid https://mui.com/x/react-data-grid/localization/#rtl-support

**Comment by LukasTy** [comment]
> Do we have a docs section about RTL?

No, we don't. 🙈 
That's a great idea! 💯 
Created an issue: https://github.com/mui/mui-x/issues/11523. 😉

**Pull request merged**